rules/scripts/bitstreams_workspace.py

```diff
@@ -472,7 +473,9 @@ def alias_lines(name, target):
             # Write substitute manifest if none came with the cache entry.
             manifest_path = os.path.join(cache_base_dir,
                                          "substitute_manifest.json")
-            self._WriteSubstituteManifest(manifest, manifest_path)
+            abs_manifest_path = os.path.join(self.cachedir,
```

Comment:
Nit: Technically, `self.cachedir` just got expanded with `os.path.expanduser`, so "absolute" isn't quite right. Maybe we should change the name here or throw in an `os.abspath` into the definition of `self.cachedir`?

Reply:
Made `self.cachedir` an absolute path.

Reply:
Brill, thanks. Looking at the new patch, I *thought* that the `abspath` call would also expand `~`, but I just checked and it doesn't! So thanks for the extra Python education :-)